src/compiler/checker.ts

```diff
@@ -7205,6 +7434,57 @@ module ts {
             return resolveCall(node, callSignatures, candidatesOutArray);
         }
 
+        function resolveDecorator(node: Decorator, candidatesOutArray: Signature[]): Signature {

```

Comment:
It's passed from getResolvedSignature, to which I added Decorator for the sake of consistency.